src/partition/repart.c

```diff
@@ -561,6 +566,12 @@ static Partition* partition_free(Partition *p) {
 
         partition_encrypted_volume_free(p->encrypted_volume);
 
+        if (p->fallback_target_for)
+                p->fallback_target_for->fallback_for = NULL;
+        else if (p->fallback_for)
```

Comment:
similar here. 

and i'd drop the `else`.

and instead add an assert here that verifies that a partition cannot be both be fallback of something and have a fallback.